src/NHibernate.Test/NHSpecificTest/NH2176/Fixture.cs

```diff
@@ -99,8 +105,42 @@ public ITransaction CreateTransaction(ISessionImplementor session)
 
 		public void EnlistInDistributedTransactionIfNeeded(ISessionImplementor session)
 		{
-			// No enlistment. This disables automatic flushes before ambient transaction
+			// No session enlistment. This disables automatic flushes before ambient transaction
 			// commits. Explicit Flush calls required.
+			// Still make sure the session connection is enlisted, in case it was acquired before 
+			// transaction scope start.
+			// Will not support nested transaction scope. (Will throw, while current NHibernate
+			// just stay in previous scope.)
+			// Will cause an "earlier than required" connection acquisition.
+			// It is required to enlist with null when the scope is ended, otherwise using
+			// the transaction without a new scope will fail by attempting to use it inside
+			// the completed scope.
+			// If an explicit transaction is ongoing, we must not enlist.
```

Comment:
Because it throws with a bunch of connection. Firebird connection seems to accept that, but not `SqlConnection`, `SQLiteConnection`, and probably a bunch of others.